prometheus/src/main/scala/filodb/prometheus/ast/Functions.scala

```diff
@@ -21,6 +21,199 @@ trait Functions extends Base with Operators with Vectors {
       throw new IllegalArgumentException(s"Invalid function name [$name]")
     }
 
+    // Below code is for validating the syntax of promql functions belonging to RangeFunctionID.
+    // It takes care of validating syntax of the tokenized input query before creating the logical plan.
+    // In case of invalid params/invalid syntax, then throw exceptions with similar error-messages like promql.
+
+    // error messages
+    val errExpectedRangeVector : String = "Expected type range_vector in call to function "
+    val errExpectedScalar : String = "Expected type scalar in call to function "
+    val errExpectedInstantVector : String = "Expected type instant vector in call to function "
+    val errExpected1Argument : String = "Expected 1 argument(s) in call to function "
+    val errExpected2Argument : String = "Expected 2 argument(s) in call to function "
+    val errExpected3Argument : String = "Expected 3 argument(s) in call to function "
+
+    /**
+     *
+     * Below function validates if the first argument to all the functions in RangeFunctionId which needs only one argument as range-expression
+     * is correct or not. If not, then throw Illegal Argument Exception with the proper error message.
+     */
+    def onlyRangeExprParamValidator(funcName : String) : Unit = {
+      // needs to have 1 parameter
+      if (allParams.size == 1) {
+        // 1st Param should be a range-vector
+        if (!allParams.head.isInstanceOf[RangeExpression])
+          throw new IllegalArgumentException(s"${errExpectedRangeVector} ${funcName}, got ${allParams.head.getClass.getSimpleName}")
```

Comment:
It's not necessary to use braces when the interpolated is just one variable.  For example: $funcName